core/src/test/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupLongPollingTest.java

```diff
@@ -167,6 +171,44 @@ void periodicCheckWhenConnectionRefused() throws Exception {
         }
     }
 
+    @Test
+    @Timeout(15)
+    void keepEndpointHealthinessWhenLongPollingTimeout() throws Exception {
+        final BlockingQueue<RequestLog> healthCheckRequestLogs = new LinkedTransferQueue<>();
+        this.healthCheckRequestLogs = healthCheckRequestLogs;
+        final Endpoint endpoint = Endpoint.of("127.0.0.1", server.httpPort());
+        try (HealthCheckedEndpointGroup endpointGroup = build(
+                HealthCheckedEndpointGroup.builder(
+                        new StaticEndpointGroup(endpoint),
+                        HEALTH_CHECK_PATH))) {
+
+            // Check the initial state (healthy).
+            assertThat(endpointGroup.endpoints()).containsExactly(endpoint);
+
+            // Drop the first request.
+            healthCheckRequestLogs.take();
+
+            // Check the endpoint is populated even after long polling timeout.
+            Thread.sleep(LONG_POLLING_TIMEOUT.toMillis());
+            assertThat(endpointGroup.endpoints()).containsExactly(endpoint);
+
+            // Must receive the '304 Not Modified' response with long polling disabled,
+            // so that the next health check respects the backoff.
+            final ResponseHeaders notModifiedResponseHeaders = healthCheckRequestLogs.take().responseHeaders();
+            assertThat(notModifiedResponseHeaders.status()).isEqualTo(HttpStatus.NOT_MODIFIED);
+            assertThat(notModifiedResponseHeaders.getLong("armeria-lphc", 1)).isEqualTo(0);
+
+            // Check the next check respected backoff, because there's no point of
+            // sending a request immediately if you get a 'not modified'.
+            final Stopwatch stopwatch = Stopwatch.createStarted();
+            healthCheckRequestLogs.take();
+            assertThat(stopwatch.elapsed(TimeUnit.MILLISECONDS))
+                    .isGreaterThan(RETRY_INTERVAL.toMillis() * 4 / 5);
+        } finally {
+            this.healthCheckRequestLogs = null;
```

Comment:
why?

Reply:
To make sure healthCheckRequestLogs is always null when the tests start?

Reply:
I thought each test uses a new instance of the test class.